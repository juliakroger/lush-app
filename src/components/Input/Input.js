import React from 'react'
import PropTypes from 'prop-types'

import styles from './Input.module.scss'

const Input = ({
  placeholder,
  value,
  valueChange,
  id,
  label,
  secondaryLabel,
  clickSecondary,
  type
}) => (
  <div className={styles.inputContainer}>
    <div className={styles.labels}>
      {label && <div className={styles.label}>{label}</div>}
      {secondaryLabel && (
        <div
          id='#secondary-label'
          className={
            clickSecondary
              ? styles.secondaryLabelClickable
              : styles.secondaryLabel
          }
          onClick={clickSecondary}
        >
          {secondaryLabel}
        </div>
      )}
    </div>
    <input
      placeholder={placeholder}
      value={value}
      onChange={e => valueChange(e.target.value)}
      id={id}
      className={styles.input}
      type={type}
      required
    />
  </div>
)

Input.propTypes = {
  placeholder: PropTypes.string,
  value: PropTypes.string.isRequired,
  valueChange: PropTypes.func.isRequired,
  id: PropTypes.string.isRequired,
  label: PropTypes.any,
  secondaryLabel: PropTypes.any,
  clickSecondary: PropTypes.func,
  type: PropTypes.string
}

Input.defaultProps = {
  placeholder: '',
  label: null,
  secondaryLabel: null,
  type: 'input'
}

export default Input
