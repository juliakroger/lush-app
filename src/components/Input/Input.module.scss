@import '../../scss/variables.scss';
@import '../../scss/responsive-utils';

@mixin secondaryLabelSimple {
  color: $grey-color;
  position: absolute;
  right: 0;
}

.inputContainer {
  padding: 10px;
  display: flex;
  flex-direction: column;
  min-width: 200px;

  .labels {
    display: flex;
    padding-bottom: 8px;
    font-size: $font-size-xsmall;
    font-weight: $font-bold;
    color: $light-black-color;
    position: relative;

    .label {
      position: relative;
      left: 0;
    }

    .secondaryLabelClickable {
      @include secondaryLabelSimple();
      cursor: pointer;
    }

    .secondaryLabelClickable:hover {
      color: $light-black-color;
    }

    .secondaryLabel {
      @include secondaryLabelSimple();
    }
  }

  .input {
    border: 1px solid $grey-color;
    color: $light-black-color;
    padding: 6px 11px;
    border-radius: 4px;
    font-size: 12px;
    line-height: $font-size-xlg;
  }
}
