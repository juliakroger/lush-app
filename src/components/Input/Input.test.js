import React from 'react'
import { shallow } from 'enzyme'
import Input from './Input'

describe('Component - Input', () => {
  it('should render without crashing', () => {
    const component = shallow(
      <Input value='value' valueChange={() => {}} id='#test-input' />
    )
    expect(component).toMatchSnapshot()
  })

  it('should render without crashing', () => {
    const component = shallow(
      <Input
        value='value'
        valueChange={() => {}}
        id='#test-input'
        placeholder='my input test'
        label='label 1'
        secondaryLabel='label 2'
      />
    )
    expect(component).toMatchSnapshot()
  })

  it('should trigger change without crashing', () => {
    let value = ''
    const changeValue = targetValue => {
      value = targetValue
    }

    const component = shallow(
      <Input
        value={value}
        valueChange={targetValue => changeValue(targetValue)}
        id='#test-input'
        placeholder='my input test'
        label='label 1'
        secondaryLabel='label 2'
      />
    )

    component.find('input').simulate('change', {
      target: { id: '#test-input', value: 'test type' }
    })

    expect(value).toEqual('test type')
    expect(component).toMatchSnapshot()
  })
})
