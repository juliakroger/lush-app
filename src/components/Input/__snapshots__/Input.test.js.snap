// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - Input should render with only required props 1`] = `
<div
  className="inputContainer"
>
  <div
    className="labels"
  />
  <input
    className="input"
    id="#test-input"
    onChange={[Function]}
    placeholder=""
    required={true}
    type="input"
    value="value"
  />
</div>
`;

exports[`Component - Input should render with placeholder, label and secondary label 1`] = `
<div
  className="inputContainer"
>
  <div
    className="labels"
  >
    <div
      className="label"
    >
      label 1
    </div>
    <div
      className="secondaryLabel"
      id="#secondary-label"
    >
      label 2
    </div>
  </div>
  <input
    className="input"
    id="#test-input"
    onChange={[Function]}
    placeholder="my input test"
    required={true}
    type="input"
    value="value"
  />
</div>
`;

exports[`Component - Input should render with secondary label, click secondary function and type 1`] = `
<div
  className="inputContainer"
>
  <div
    className="labels"
  >
    <div
      className="secondaryLabelClickable"
      id="#secondary-label"
      onClick={[MockFunction]}
    >
      label 2
    </div>
  </div>
  <input
    className="input"
    id="#test-input"
    onChange={[Function]}
    placeholder=""
    required={true}
    type="password"
    value="value"
  />
</div>
`;

exports[`Component - Input should trigger input change 1`] = `
<div
  className="inputContainer"
>
  <div
    className="labels"
  >
    <div
      className="label"
    >
      label 1
    </div>
    <div
      className="secondaryLabel"
      id="#secondary-label"
    >
      label 2
    </div>
  </div>
  <input
    className="input"
    id="#test-input"
    onChange={[Function]}
    placeholder="my input test"
    required={true}
    type="input"
    value=""
  />
</div>
`;
